### PR TITLE
deployment/docker: add grafana datasource to the docker-compose files

### PR DESCRIPTION
**Pull request by dmitryk-dk**
Add [VictoriaMetrics datasource](https://github.com/VictoriaMetrics/grafana-datasource) to the docker-compose environment.

I have separated environments for data sources and the default setup. I used the docker compose merge functionality, and it helps to avoid many tricks via bash

Related issue: https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3920, https://github.com/VictoriaMetrics/grafana-datasource/issues/113

### CODE REVIEW

**Comment by hagen1778** [comment]
@dmitryk-dk is it possible to use something similar to init container - a script which will download and install the plugin as we recommend doing here https://github.com/VictoriaMetrics/grafana-datasource#install-in-kubernetes ?

**Comment by dmitryk-dk** [comment]
> @dmitryk-dk is it possible to use something similar to init container - a script which will download and install the plugin as we recommend doing here https://github.com/VictoriaMetrics/grafana-datasource#install-in-kubernetes ?

As far as I know, Docker doesn't have that concept as Kubernetes has. So it would be not possible to make it in that manner.

Maybe @zekker6 knows more about it.

**Comment by Haleygo** [comment]
> > @dmitryk-dk is it possible to use something similar to init container - a script which will download and install the plugin as we recommend doing here https://github.com/VictoriaMetrics/grafana-datasource#install-in-kubernetes ?
> 
> As far as I know, Docker doesn't have that concept as Kubernetes has. So it would be not possible to make it in that manner.
> 
> Maybe @zekker6 knows more about it.

Yeah, for docker, we can override images's default entrypoint and cmd with our script.
Could be something like
```
services:
  grafana:
    container_name: grafana
    image: grafana/grafana:latest
    ports:
      - 3000:3000
    entrypoint: ["/bin/bash", "-c"]
    command: ["chmod +x /download-vm-datasourcet.sh && /download-vm-datasource.sh && sleep 1000"]
    volumes:
      - ./download-vm-datasource.sh:/download-vm-datasource.sh
```
 
ps. we should also upgrade grafana image to [9.5.0+](https://hub.docker.com/layers/grafana/grafana/9.5.0/images/sha256-258a4c283403a26848bd6a9abb6b6867d21ec83ca228d14da4e66589a49676ef?context=explore) if we use this way, since it [gained `curl`](https://github.com/grafana/grafana/pull/65547) about that time.

**Comment by dmitryk-dk** [comment]
> > > @dmitryk-dk is it possible to use something similar to init container - a script which will download and install the plugin as we recommend doing here https://github.com/VictoriaMetrics/grafana-datasource#install-in-kubernetes ?
> > 
> > 
> > As far as I know, Docker doesn't have that concept as Kubernetes has. So it would be not possible to make it in that manner.
> > Maybe @zekker6 knows more about it.
> 
> Yeah, for docker, we can override images's default entrypoint and cmd with our script in [docker-compose yaml](https://docs.docker.com/compose/compose-file/05-services/#entrypoint). Could be something like
> 
> ```
> services:
>   grafana:
>     container_name: grafana
>     image: grafana/grafana:latest
>     ports:
>       - 3000:3000
>     entrypoint: ["/bin/bash", "-c"]
>     command: ["chmod +x /downloadvmdatasource.sh && /downloadvmdatasource.sh && /run.sh"]
>     volumes:
>       - ./downloadvmdatasource.sh:/downloadvmdatasource.sh
> ```
> 
> ps. we should also upgrade grafana image to [9.5.0+](https://hub.docker.com/layers/grafana/grafana/9.5.0/images/sha256-258a4c283403a26848bd6a9abb6b6867d21ec83ca228d14da4e66589a49676ef?context=explore) if we use this way, since it [gained `curl`](https://github.com/grafana/grafana/pull/65547) about that time.

Hi @Haleygo ! I have simplified logic and added the download script as you suggested. can you take a look please.

I tested it on my local env it works well

**Comment by Haleygo** [comment]
Looking good!
Little proposal, how about switching them and making victoriametrics datasource as the default one with `docker-cluster-up`?

**Comment by dmitryk-dk** [comment]
> Looking good! Little proposal, how about switching them and making victoriametrics datasource as the default one with `docker-cluster-up`?

We splitted realisation for two envs specially, because our plugin is unsigned and some of the users do not want to use it. So they can choose what of the datasource to use.

File: deployment/docker/docker-compose-cluster-with-datasource.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+version: "3.5"
```
[the remark this answers is not in the document]
Comment (dmitryk-dk) [reply deployment/docker/docker-compose-cluster-with-datasource.yml]: Hi @hagen1778 ! I have moved the files related to the Victoriameterics data source to own folder. In the https://github.com/VictoriaMetrics/grafana-datasource/pull/119, I added information on how to `up` and `down`

**Review by hagen1778 — APPROVED** [review]
LGTM

**Pull request merged**